xtask/src/main.rs

```diff
@@ -0,0 +1,364 @@
+use anyhow::{anyhow, Context, Result};
+use clap::{Parser, Subcommand};
+use std::env;
+use std::path::{Path, PathBuf};
+use std::process::{Command, Stdio};
+
+#[derive(Parser)]
+#[command(name = "xtask")]
+#[command(about = "Development tasks for rs-matter ConnectedHomeIP integration")]
+struct Cli {
+    #[command(subcommand)]
+    command: Commands,
+}
+
+#[derive(Subcommand)]
+enum Commands {
+    /// Setup ConnectedHomeIP environment for integration testing
+    ItestSetup {
+        /// ConnectedHomeIP repository reference (branch/tag/commit)
+        #[arg(long, default_value = "master")]
+        connectedhomeip_ref: String,
+        /// Force rebuild even if cached
+        #[arg(long)]
+        force_rebuild: bool,
+    },
+    /// Build rs-matter examples
+    Build {
+        /// Build profile (debug or release)
+        #[arg(long, default_value = "debug")]
+        profile: String,
+        /// Additional cargo features
+        #[arg(long)]
+        features: Vec<String>,
+    },
+    /// Run ConnectedHomeIP YAML integration tests
+    Itest {
+        /// Test names to run (if empty, runs all enabled tests)
+        tests: Vec<String>,
+        /// Build profile to use for rs-matter examples
+        #[arg(long, default_value = "debug")]
+        profile: String,
+        /// Timeout for each test in seconds
+        #[arg(long, default_value = "120")]
+        timeout: u64,
+        /// Skip building rs-matter (assume it's already built)
+        #[arg(long)]
+        skip_build: bool,
+    },
+}
+
+const DEFAULT_FEATURES: &[&str] = &["os", "rustcrypto", "log"];
+
+#[tokio::main]
+async fn main() -> Result<()> {
+    let cli = Cli::parse();
+
+    match &cli.command {
+        Commands::ItestSetup { connectedhomeip_ref, force_rebuild } => {
+            setup_connectedhomeip(connectedhomeip_ref, *force_rebuild).await
+        }
+        Commands::Build { profile, features } => {
+            build_rs_matter(profile, features).await
+        }
+        Commands::Itest { tests, profile, timeout, skip_build } => {
+            run_integration_tests(tests, profile, *timeout, *skip_build).await
+        }
+    }
+}
+
+async fn setup_connectedhomeip(connectedhomeip_ref: &str, force_rebuild: bool) -> Result<()> {
+    let project_root = get_project_root()?;
+    let connectedhomeip_dir = project_root.join("connectedhomeip");
```

Comment:
I don't feel very good with having "connectedhomeip" directly in the project root.
how about placing "connectedhomeip" as follows:
`.build/itest/connectedhomeip"?

This way we should ".gitignore" all of `.build` and then in future we can use this place for placing other stuff too.